libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -884,15 +918,27 @@ bool HTTPClient::sendHeader(const char * type)
 
     String header;
     // 128: Arbitrarily chosen to have enough buffer space for avoiding internal reallocations
-    header.reserve(_headers.length() + _uri.length() +
-            _base64Authorization.length() + _host.length() + _userAgent.length() + 128);
+    header.reserve(_headers.length() + _uri->path.length() +
+            _uri->base64Authorization.length() + _uri->host.length() + _userAgent.length() + 128);
     header += type;
     header += ' ';
-    if (_uri.length()) {
-        header += _uri;
+
+    String target;
+    if (_useProxy) {
+        // If using proxy, use full uri in request line (http://foo:123/bar)
+        target = _uri->protocol + F("://") + _uri->host;
```

Comment:
When a `String` is `::reserve(sufficient_space)`, and to avoid ram fragmentation, it is generally preferable to do
```cpp
target = _uri->protocol;
target += F("://");
target += _uri->host;
```
to avoid use of temporaries when building the pieces.